### ISSUE TITLE
Add a local backup feature for save data

### ISSUE THREAD

**Issue by felixlang** [CryoByte33/steam-deck-utilities#73]
This is probably a longshot to implement but I wanted to mention it anyway, since it would be fixing one of the worst aspects of the Steam Deck currently.

Something you touched on in your 2.0 release video is that deleting game data will delete save data for games that don't have cloud backup support (at least for games played through Proton). I wonder if it would be possible to add a tool that makes a local backup of all games' save and configuration files.

If it's possible to do this, then I imagine it might require identifying the installation files from the games' manifests, possibly by querying SteamDB (ex. [Rayman Origins](https://steamdb.info/depot/207491/)), along with Windows files and dependencies. Then the tool could create backups of all files that aren't part of any of those lists.

It's probably not worth the amount of work that would be needed, but if I'm wrong then it would certainly be a big improvement to the Deck, as this is the only console I'm aware of that routinely deletes save data when uninstalling games.

**Comment by CryoByte33** [comment]
Hmmm not a bad idea, I'd already planned a "data backup" tool, but having it very granular and easily restorable could be difficult.

Would [OpenCloudSaves](https://github.com/DavidDeSimone/OpenCloudSaves) be a better fit for what you describe?

**Comment by ElNovi** [comment]
We can identify which game has Steam Cloud Saves by visiting this url https://store.steampowered.com/api/appdetails?appids=1649240 (this is for Returnal) Steam Cloud is category id **23**

Rather then creating backups, can we just not delete the save files for such games? I don't know if there is a way to identify save files.

**Comment by CryoByte33** [comment]
Yeah, but would that be a benefit? If Steam's already handling the saves, is there a use case for getting a local copy to, since you can just use a browser to download your save files?

In addition, I'm not sure if we can use Steam's manifest for "save game locations", but that definitely requires investigation if we want to do this.

**Comment by ElNovi** [comment]
> In addition, I'm not sure if we can use Steam's manifest for "save game locations", but that definitely requires investigation if we want to do this.

I will take a look at it when i have a time. Avoiding deleting saves is by me the best solution, cos we want to delete just shaders anyway right? For backups there is already solution as you mentioned https://github.com/DavidDeSimone/OpenCloudSaves

**Comment by CryoByte33** [comment]
> Avoiding deleting saves is by me the best solution, cos we want to delete just shaders anyway right?

Yeah, my main concern is the expectation that things will be "proper" after cleanup. As backwards as it sounds, having users more aware of the data being cleaned up might be much better, since we can't be held accountable for missing data.

Also, if we "save" the correct files, games may fail to launch after reinstall or boot. Steam expects either a "correct" compatdata directory, or none at all. Having one with files present will likely break quite a bit, and might result in data loss anyway.

**Comment by ElNovi** [comment]
> > Avoiding deleting saves is by me the best solution, cos we want to delete just shaders anyway right?
> 
> Yeah, my main concern is the expectation that things will be "proper" after cleanup. As backwards as it sounds, having users more aware of the data being cleaned up might be much better, since we can't be held accountable for missing data.
> 
> Also, if we "save" the correct files, games may fail to launch after reinstall or boot. Steam expects either a "correct" compatdata directory, or none at all. Having one with files present will likely break quite a bit, and might result in data loss anyway.

Agree. Would not deleting compat data for games that don't have Steam Cloud Saves turned on be good compromise?

**Comment by CryoByte33** [comment]
> Agree. Would not deleting compat data for games that don't have Steam Cloud Saves turned on be good compromise?

It could be, but then half of the benefit of "delete game data" is lost, since a large amount of the reason it exists is to free space consumed by dangling Steam games 🤔

**Comment by ElNovi** [comment]
> It could be, but then half of the benefit of "delete game data" is lost, since a large amount of the reason it exists is to free space consumed by dangling Steam games 🤔

As i see it, it is either this or popup warning to backup data first. The tools for backup already exists and it would be too much of a hassle to implement it into Cryoutils ... Those save files seems to be all over the place depending if the games run through proton/native/etc ...

Besides i have more then 800 games on Steam and i don't think that i ever lost saves nor i do think that i played game without Steam Cloud Saves. If the game doesn't support Steam Cloud Saves it has it's own Cloud Save Sync like for example Disney Dreamlight Valley ...
